Support Python 2 and 3

pmclanahan opened the pull request:
First attempt at simultaneous py2 and py3 support.

Osmose commented on product_details/storage.py:
```diff
@@ -173,14 +175,14 @@ def update(self, name, content, last_modified):
                 name, self.json_dir))
             tf = tempfile.NamedTemporaryFile(delete=False)
-            tf.write(content)
+            tf.write(binary_type(content.encode('utf8')))
```
Is `binary_type` necessary here? From what I just read, in Python 3 encode returns a `bytes` object already, and in Python 2 it returns an 8-bit `str`.

pmclanahan replied:
Looks like you're right. Thanks.

Osmose commented on product_details/version_compare/__init__.py:
```diff
@@ -178,8 +181,8 @@ def simplify_version(version):
     # stringify
-    pieces = map(str, pieces)
-    suffixes = map(str, suffixes)
+    pieces = [str(x) for x in pieces]
```
![](http://i.imgur.com/cYimkuT.gif)

Osmose commented on tox.ini:
```diff
@@ -1,17 +1,21 @@
-envlist =
-    py27-{1.7,1.8}
+skip_missing_interpreters = true
+envlist = py{27,33,34,py}-dj{17,18},py35-dj18,flake8
```
Now that it's been almost 2 months, you should add Django 1.9 here as well. :D

pmclanahan replied:
Good point :)

pmclanahan replied:
Turns out this is more than just adding 1.9 to the tox file. I'll fix this in a subsequent PR.

Osmose commented:
r+, this looks great to me, and the tests all pass locally for me. Nice work!

pmclanahan commented:
Updated some things from review, and found an error with how I was using urllib from six in the management command. I think all's well now.

Oh, and I updated the release.sh script to just build the sdist and wheel and recommend using `twine` to upload to PyPI.
